src/functional/ffront/foast_passes/type_deduction.py

```diff
@@ -348,6 +348,7 @@ def _visit_neighbor_sum(self, node: foast.Call, **kwargs) -> foast.Call:
 
     def _visit_broadcast(self, node: foast.Call, **kwargs) -> foast.Call:
         field_type = cast(ct.FieldType, node.args[0].type)
+        assert isinstance(node.args[1], foast.TupleExpr)
         broadcast_dims_expr = node.args[1].elts
```

Comment:
```suggestion
        broadcast_dims_expr = cast(node.args[1], foast.TupleExpr).elts
```
This is ensured by `is_callable` function and we do not place any asserts that fail anyway if an invariant is violated.

Reply:
But the Mypy (and thus the pre-commit hooks and thus CI) fails without this.

Reply:
https://github.com/GridTools/gt4py/runs/6706727049?check_suite_focus=true

Reply:
I would be really surprised if this fails after you have added the cast that I suggested. Am I missing something here?

Reply:
Ah sorry I somehow missed the suggestion!